diagnostics/monitoring_server.rs

```diff
@@ -0,0 +1,90 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{
+    convert::Infallible,
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+    thread,
+};
+
+use futures::executor::block_on;
+use hyper::{
```

Comment:
A use the [hyper](https://crates.io/crates/hyper) crate which seems to be one of the most popular and mature crates for creating REST servers (I actually haven't tried it as a client crate for `reporter`, may pay attention to it if needed).
This crate requires using a `tokio` runtime.